### PR TITLE
Add matrix build for supported targets

### PR DESCRIPTION
**Pull request by paulbartell**
Description
-----------
Split builds into a separate GitHub workflow.
Build for all supported targets with multiple esp-idf versions.

Test Steps
-----------

Checklist:
----------

- [ ] I have tested my changes. No regression in existing tests.
- [ ] I have modified and/or added unit-tests to cover the code changes in this Pull Request.

Related Issue
-----------

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

**Review by n9wxu — DISMISSED** [review]

File: .github/workflows/build.yml [hunk 1]
```diff
@@ -0,0 +1,56 @@
...
+          git submodule update --checkout --init --recursive
+          echo -e "::endgroup::"
+          echo -e "${{ env.bashPass }} ${{ env.stepName }} ${{ env.bashEnd }}"
+
+      - name: Build for
+        uses: espressif/esp-idf-ci-action@v1
+        with:
+          esp_idf_version: ${{ matrix.esp_idf_version }}
+          target: ${{ matrix.esp_target }}
+          path: './'
```
Comment (AniruddhaKanhere) [inline .github/workflows/build.yml:56]: No new line?

**Review by AniruddhaKanhere — APPROVED** [review]
Approved with one minor comment about missing new line at EOF.

**Review by archigup — DISMISSED** [review]

File: .github/workflows/build.yml (hunk 1 above)
Comment (paulbartell) [reply .github/workflows/build.yml]: Fixed.

File: .github/workflows/build.yml [hunk 2]
```diff
@@ -0,0 +1,56 @@
+name: "CI Checks"
```
Comment (Skptak) [inline .github/workflows/build.yml:1]:
```suggestion
name: "Demo Build Checks"
```

Comment (Skptak) [reply .github/workflows/build.yml]: Just so we don't have the same name for the jobs :)

File: .github/workflows/build.yml [hunk 3]
```diff
@@ -0,0 +1,56 @@
...
+    branches: ["**"]
+  pull_request:
+    branches: [main]
+  workflow_dispatch:
+
+jobs:
+  build:
+    runs-on: ubuntu-latest
+    strategy:
+      matrix:
```
Comment (Skptak) [inline .github/workflows/build.yml:20]:
```suggestion
    strategy:
      fail-fast: false
      matrix:
```

Comment (Skptak) [reply .github/workflows/build.yml:20]: Allow all jobs to run even if a single one fails

Comment (Skptak) [reply .github/workflows/build.yml:20]: https://docs.github.com/en/actions/using-jobs/using-a-matrix-for-your-jobs#handling-failures for more info

Comment (Skptak) [reply .github/workflows/build.yml:20]: https://github.com/FreeRTOS/iot-reference-esp32c3/actions/runs/7216197357 example of diff

**Review by AniruddhaKanhere — APPROVED** [review]

**Review by Skptak — APPROVED** [review]

**Pull request merged**